### ISSUE TITLE
knit("file.ipynb")

### ISSUE THREAD

**Issue by yihui** [yihui/knitr#1619]
Given that we have the wonderful **reticulate** package to support Python and **RJulia** to run Julia code, I think it is good time to support knitting Jupyter notebooks now.

**Comment by davidbody** [comment]
I just saw https://github.com/rstudio/rmarkdown/commit/67b7f5fc779e4cfdfd0f021d3d7745b6b6e17149 so I'm not sure you need these, but for future reference:

[The Jupyter Notebook Format](https://nbformat.readthedocs.io/en/latest/index.html)

Jupyter notebook examples:

https://gist.github.com/davidbody/cd375b92bb9bed81c92257dcd7efd638

**Comment by yihui** [comment]
Yes, I need more examples. I don't use Jupyter myself, so I really appreciate your examples! I only spent a morning on writing that function, and I have no idea it could actually be useful to any Jupyter users at all.

**Comment by grst** [comment]
[jupytext](https://github.com/mwouts/jupytext) robustly converts `Rmd` into `ipynb` and vice-versa. 
I actually use it in production for quite a while now to convert my jupyter notebooks to Rmd and render them with `knitr`. The only thing that does not work are jupyter cell magics, but I can live with that. 

The author, @mwouts, is really interested in improving the compatiblity between the python and R ecosystems, so feel free to contact him if you have any questions.

**Comment by mwouts** [comment]
> The author, @mwouts, is really interested in improving the compatiblity between the python and R ecosystems

Sure! And we do have some people interested in that on the Python side, see for instance https://github.com/mwouts/jupytext/issues/267 where we test the idea of running a complex Jupyter notebook in RStudio.

@yihui , please let me know if you need help on `jupytext`. I think you may want to give a try to either the [command line version](https://jupytext.readthedocs.io/en/latest/using-cli.html), or to the [Python API](https://jupytext.readthedocs.io/en/latest/using-library.html), probably a good fit when used into `reticulate`. For instance, reading an `ipynb` file, and returning the corresponding Rmd file as text is
```
nb = jupytext.read('notebook.ipynb')
rmd_content = jupytext.writes(nb, fmt='Rmd')
```

You may also like the fact that one can _update_ the text inputs in the `ipynb` file from the Rmd representation: `jupytext --to ipynb --update notebook.md`.

> The only thing that does not work are jupyter cell magics

Cell magics are IPython-specific. By default Jupytext comment them out in the R Markdown representation, as I don't think `reticulate` supports them.

I would also add that interactive (javascript) outputs do not work in reticulate, at least they did not the last time I gave it a try (https://github.com/rstudio/reticulate/issues/269).

**Comment by yihui** [comment]
Revisiting my own request: I no longer think it's worthwhile to build `.ipynb` support into knitr, given the conversion tools already mentioned above. Closing.
